api/analyzedphenotypes.materializedview.api.inc

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * @file
+ *
+ * API: MATERIALIZED VIEWS.
+ * Manage property and operation pertaining to materizalized view.
+ */
+
+
+/**
+ * @TODO: Replace this function with the function below.
+ */
+function analyzedphenotypes_mview($property, $parameter = null) {
+  $mviewquery  = ap_construct_mviewquery();
+  $mviewschema = ap_construct_mviewschema();
+
+  return array(
+    'query' => $mviewquery,
+    'schema' => $mviewschema,
+  );
+}
+
+/**
+ * @TODO: Replace this function with the function below.
+ */
+function analyzedphenotypes_updatemview() {
+  ap_update_mview();
+}
+
+
+///////
+
+
+/**
+ * Prepare SQL query required by materialized view.
+ * Dependencies: ap_get_variablenames() in systemvars.api.
+ *
+ * @return
+ *   Text block, sql query statement.
+ */
+function ap_construct_mviewquery() {
```

Comment:
what about ap_construct_mview_phenotype_query() to make the function name specific to the mview it's constructing a query for.